docs/shared/packages/powerpack-conformance/powerpack-conformance-plugin.md

```diff
@@ -1,6 +1,198 @@
+| depConstraints                                   | _Array<object>_           | _[]_    | List of dependency constraints between projects                                                                                                                                                                                                |
+
+#### Dependency constraints
+
+The `depConstraints` is an array of objects representing the constraints defined between source and target projects. A
+constraint must include `sourceTag` or `allSourceTags`. The constraints are applied with **AND** logical operation - for
+a given `source` project the resulting constraints would be **all** that match its tags.
+
+| Property                 | Type            | Description                                                                        |
+| ------------------------ | --------------- | ---------------------------------------------------------------------------------- |
+| sourceTag                | _string_        | Tag that source project must contain to match the constraint                       |
+| allSourceTags            | _Array<string>_ | List of tags the source project must contain to match the constraint               |
+| onlyDependOnLibsWithTags | _Array<string>_ | The source **can depend only** on projects that contain at least one of these tags |
+| notDependOnLibsWithTags  | _Array<string>_ | The source **can not depend** on projects that contain at least one of these tags  |
+
+### Ensure Owners
+
+This rule requires every project to have an owner defined for the [`@nx/powerpack-owners` plugin](/nx-api/powerpack-owners)
+
+Set the `rule` property to: `@nx/powerpack-conformance/ensure-owners`
+
+```json {% fileName="nx.json" %}
+{
+  "conformance": {
+    "rules": [
+      {
+        "rule": "@nx/powerpack-conformance/ensure-owners"
+      }
+    ]
+  }
+}
+```
+
+## Custom Conformance Rules
+
+To write your own conformance rule, specify a relative path to a typescript file as the rule name:
+
+```json {% fileName="nx.json" %}
+{
+  "conformance": {
+    "rules": [
+      {
+        "rule": "./tools/local-conformance-rule.ts"
+      }
+    ]
+  }
+}
+```
+
+The rule definition file should look like this:
+
+```ts {% fileName="tools/local-conformance-rule.ts" %}
+import type {
+  ConformanceRule,
+  ConformanceRuleResult,
+} from '@nx/powerpack-conformance';
+
+const rule: ConformanceRule<any> = {
+  name: 'local-conformance-rule-example',
+  category: 'security', // `consistency`, `maintainability`, `reliability` or `security`
+  reporter: 'project-reporter', // `project-reporter` or `project-files-reporter`
+  implementation: async (context): Promise<ConformanceRuleResult> => {
+    const { projectGraph, ruleOptions } = context;
+    // Your rule logic goes here
+    return {
+      severity: 'low', // 'high', 'medium' or 'low'
```

Comment:
You could note that commonly the severity of a rule should be relatively static based on what the rule is trying to achieve, but it is part of the rule's returned result and not the rule metadata to give rule authors the flexibility to customize the severity if the rule checks for a number of different cases within its implementation